firebase-firestore/src/test/java/com/google/firebase/firestore/remote/BloomFilterTest.java

```diff
@@ -0,0 +1,178 @@
+    BloomFilter bloomFilter1 = new BloomFilter(new byte[1], 0, 1);
+    assertFalse(bloomFilter1.isEmpty());
+    BloomFilter bloomFilter2 = new BloomFilter(new byte[1], 7, 1);
+    assertFalse(bloomFilter2.isEmpty());
+  }
+
+  @Test
+  public void testNonEmptyBloomFilterThrowException() {
+    IllegalArgumentException negativePaddingException =
+        assertThrows(IllegalArgumentException.class, () -> new BloomFilter(new byte[1], -1, 1));
+    assertThat(negativePaddingException).hasMessageThat().contains("Invalid padding: -1");
+    IllegalArgumentException overflowPaddingException =
+        assertThrows(IllegalArgumentException.class, () -> new BloomFilter(new byte[1], 8, 1));
+    assertThat(overflowPaddingException).hasMessageThat().contains("Invalid padding: 8");
+
+    IllegalArgumentException negativeHashCountException =
+        assertThrows(IllegalArgumentException.class, () -> new BloomFilter(new byte[1], 1, -1));
+    assertThat(negativeHashCountException).hasMessageThat().contains("Invalid hash count: -1");
+    IllegalArgumentException zeroHashCountException =
+        assertThrows(IllegalArgumentException.class, () -> new BloomFilter(new byte[1], 1, 0));
+    assertThat(zeroHashCountException).hasMessageThat().contains("Invalid hash count: 0");
+  }
+
+  @Test
+  public void testBloomFilterProcessNonStandardCharacters() {
+    // A non-empty BloomFilter object with 1 insertion : "ÀÒ∑"
+    BloomFilter bloomFilter = new BloomFilter(new byte[] {(byte) 237, 5}, 5, 8);
+    assertTrue(bloomFilter.mightContain("ÀÒ∑"));
+    assertFalse(bloomFilter.mightContain("Ò∑À"));
+  }
+
+  @Test
+  public void testEmptyBloomFilterMightContainAlwaysReturnFalse() {
+    BloomFilter bloomFilter = new BloomFilter(new byte[0], 0, 0);
+    assertFalse(bloomFilter.mightContain("abc"));
+  }
+
+  @Test
+  public void testBloomFilterMightContainOnEmptyStringAlwaysReturnFalse() {
+    BloomFilter emptyBloomFilter = new BloomFilter(new byte[0], 0, 0);
+    BloomFilter nonEmptyBloomFilter =
+        new BloomFilter(new byte[] {(byte) 255, (byte) 255, (byte) 255}, 1, 16);
+
+    assertFalse(emptyBloomFilter.mightContain(""));
+    assertFalse(nonEmptyBloomFilter.mightContain(""));
+  }
+
+  /**
+   * Golden tests are generated by backend based on inserting n number of document paths into a
+   * bloom filter.
+   *
+   * <p>Full document path is generated by concatenating documentPrefix and number n, eg,
+   * projects/project-1/databases/database-1/documents/coll/doc12.
+   *
+   * <p>The test result is generated by checking the membership of documents from documentPrefix+0
+   * to documentPrefix+2n. The membership results from 0 to n is expected to be true, and the
+   * membership results from n to 2n is expected to be false with some false positive results.
+   */
+  @Test
+  public void testBloomFilterGoldenTest() throws Exception {
+    String documentPrefix = "projects/project-1/databases/database-1/documents/coll/doc";
```

Comment:
I'd like to create one test method per golden test. Otherwise, the first golden test failure will hide any other failures from any other tests. I think the test way to do this would be to create a helper method like "runGoldenTest(String name)" and then a bunch of `@Test` methods that just call the runGoldenTest() method with the name of the test, which will be the file name.